### ISSUE TITLE
[EN][TALENT][SHAMAN] Maelstrom Weapon PPM

### ISSUE THREAD

**Issue by Tetuz** [Atlantiss/BugTracker#9457]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

[//]: # (On the Bugtracker all titles of reports must be written in English so that everyone, no matter which language they use everyday, could search and check if the problem already has been reported by another player. When describing the problem you can use English as well as Polish and in nearest future - others. It is important to determine at the beginning of a title in which language it is written by using symbols [EN] or [PL].)

[//]: # (Enclose links to things related to the bug using http://wowhead.com or http://cata.openwow.com)
[//]: # (Write your tickets according to the format:)
[//]: # ([EN][Quest][Zone] A Vision of the Past)
[//]: # ([EN][NPC] Thoralius the Wise)
[//]: # ([EN][Spell][Class] Frostfire Bolt)
[//]: # ([PL][Talent][Class] Brain Freeze)
[//]: # ([EN][Glyph][Class] Glyph of Frostfire Bolt)
[//]: # ([EN][Npc][Drop] Thoralius the Wise)
[//]: # ([EN][Web] Armory doesnt work)

**Description**: Hello, i think maelstrom weapon procs are a bit low, it seems to be a problem on a lot of private servers, so i tested here and it seems to be a bit low (not terribly but still it's low).

**Current behaviour**: Done a 10 minute test with wow in game stopwatch, just with autoattack and casting lighting bolt when i got 5 stacks of maelstrom weapon (except one, got distracted and use one on 3-4 stacks), without any haste buff but with gear; i got 19,74 PPM according to my calculation (40 lightining bolt, meaning 40*5=200 procs, (200procs*60)/608s= 19,74 Procs Per Minute)
![wowscrnshot_082517_144730](https://user-images.githubusercontent.com/26360436/29734554-251fc33c-89f3-11e7-9df2-daa95132426c.jpg)

**Expected behaviour**: Maelstrom Weapon should have 10 PPM with one weapon, when dual wielding it should be 25PPM ( don't know why, if someone can explain would be great)
Sources:
http://wowwiki.wikia.com/wiki/Maelstrom_Weapon Only proc reduction was in WoD, so it should be safe to say PPM was the same in wotlk, cata and mop (?)
http://forum.warmane.com/showthread.php?t=228239 
https://www.freakz.ro/forum/FIXED-Shaman-Enhancement-Passive-Maelstrom-weapon-t348732.html I know it's from MoP but as i said PPM should be the same, anyway there's some simcraft coding, it may be useful
http://blue.mmo-champion.com/topic/16360-cataclysm-shaman-faq/

**Steps to reproduce the problem**:

1. take off weapon imbues, haste proc or other haste sources
2. Hit dummy with autoattacks and use lightning bolt 
3. Check number of procs and time spent attacking

**Revision**: 5480

**Comment by Psychobird7** [comment]
Hi! 

The freakz forum links to a spell ID that does not exist. 
51532 is not a spell in Cataclysm.   
The spell IDs used by Maelstrom Weapon (which I will call MSW here) are 51528 (Rank 1 talent), 51529 (Rank 2), 51530 (Rank 3), and 53817 (Buff). 

The Warmane people seem to not understand how PPM works. 

The MMOC forum post being the only reliable source, and since I have little knowledge of Enhancement Shaman mechanics, I will assume the PPM for MSW to be 9 in the following.  
As a side note, dual-wielding does *not* change PPM. 


For clarity, PPM functions as follows:  
PPH is the chance for each hit (melee swing, melee ability, etc) to trigger a talent (in this case, MSW). 
PPH = (WSpeed) / (60 / PPM) 

Applied to MSW's 9 PPM for Enhancement Shamans, this amounts to: 
PPH = 2.6 / (60 / 9) = 0.39 
--> 39% chance for each hit to trigger MSW. 

**PPM itself is not modified by Haste, talents or anything else.** This means Flurry does not hinder it. 
Haste only makes you attack faster, which gives you more chances to proc. 
-- 25 PPM would equal a 108% proc chance, which is obviously not possible, hence my comment about the Warmane forum people not knowing what they're talking about. 


Your testing shows a total of 530 Swings and 40 LB casts. Assuming you only used them at 5 stacks (I know you did not), this makes it 200 MSW procs, or a 37.7% proc rate per attack. 
Taking into account the 4 stacks you wasted, that is 204 procs, or a 38.4% proc rate. 

Simcraft, with a profile modified to only auto-attack -- and use LB when it reaches 5 stacks of MSW -- for 6min on a Patchwerk fight, showed 388 Swings, 156 MSW procs, and 37 LB casts. 
This amounts to 40.2% proc per attack.  

The same profile, using the same parameters, and a more precise simulation (500k iterations), for 10min this time, showed 636 Swings, 257 MSW procs, and 49 LB casts. 
This is a 40.4% MSW proc on each swing. 

While Simcraft code seems to show 10 PPM (I checked), the proc rate obtained on simulations are not consistent with such a number. 
9 PPM amounts to 39% per attack, 10 PPM would be 43.3%. 


--> Based on said MMOC post, the proc rates you obtained are close to correct.  
If you were to use abilities while testing, you would see a significant increase in procs. 


TL;DR: *If* the information provided on the MMOC thread is correct (which is to be assumed, considering a Blizzard employee gave a positive comment on it), and MSW is indeed intended to be set at 9 PPM, your testing did not show a statistically significant difference in procs, and it seems MSW is working as -- or close to -- intended. 
You were probably unlucky with procs, which is very plausible considering your small sample size.

**Comment by Kaityz** [comment]
Psychobird7, auto-attack misses can't proc maelstrom, but in your calculations you count them as well in total swings, which is probably wrong. I think it should be:

200 MSW procs / (530 total swings - 41 misses) = 0.408 or 40.8% proc chance

The same with your simulation, but i don't know the number of misses.

**Comment by Tetuz** [comment]
Hi Guys, thanks for your detailed answers, just what i neeed.
@Psychobird7 I think base PPM should be 9-10 (rank 5 of msw in wotlk gave 10 ppm, since there was no reduction it should stay the same in cata, other sources claiming the same thing: http://www.wowhead.com/forums&topic=175441/maelstrom-weapon-proc , http://www.mmo-champion.com/threads/811901-does-maelstrom-weapon-have-an-internal-cooldown , https://us.battle.net/forums/en/wow/topic/3341764314 alkari's comment). And as said in blue PPM kinda stacks with haste, it doesn't change the base value, whick gives us like 39% per hit chance to gain a stack, but hitting more time in a minute i'll get more than 9 stacks.
" - MW ranks give 3/6/9 base procs per minute (PPM).
- Note that this is before dualwielding, windfury, instant attacks, and haste like flurry or WF totem. MW procs a lot more than 9 per minute. Due to instant attacks, slow weapons are better than fast. "
 
I think the 25 PPM which wowwiki refers to when dual wielding aren't BASE PPM, that being said i don't know how to test if dual wielding base ppm is correct.
 On my test i got something like 198-199 procs, because as i said i casted a lb when i was at 3-4 stacks but i don't think that will be a huge difference, anyway i'll try to test in some other ways.

@Kaityz Having a 6,6% chance to miss on my target (and don't remembering the number of missis, i'll check recount when i log in) we can multiply my ppm for that chance to have a slightly more accurate number: 19,74*93,6% = 21,08, which according to wowwiki it's still a bit low (considering i had gear on me, which gives me a bit of haste)
